src/handlers/channel.ts

```diff
@@ -422,6 +421,49 @@ export async function editChannel(
   );
 }
 
+export async function editChannelOverwrite(
+  channelID: string,
+  overwrite: Overwrite,
+  reason?: string,
+) {
+  const hasManageChannelPerm = await botHasChannelPermissions(
```

Comment:
This needs a little bit of research as im not 100% certain but if i had to guess, im sure Discord would not let a bot give/remove permission from another role/member if itself did not have the permissions being edited.

Also can you double check MANAGE_ROLES is not required? I think thats the MANAGE Permissions one and im almost positive that might be required

Reply:
Found out that permissions are little more complex. Other functions are facing the same problem too.